src/components/container/bookmark.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Container } from 'reactstrap';
+import { withToastManager } from 'react-toast-notifications';
+import { ArticleView } from '../singleArticle';
+import NavBar from '../functional/navBar';
+import { getAllBookmarks, removeBookmark } from '../../redux/action-creators/bookmark';
+
+class Bookmark extends Component {
+  componentDidMount() {
+    const { history } = this.props;
+    if (!localStorage.token) history.push('/');
+    const { allBookmarks } = this.props;
+    allBookmarks();
+  }
+
+  deleteBookmark = async (slug) => {
+    const { remove } = this.props;
+    await remove(slug);
+    const { toastManager, bookmark } = this.props;
```

Comment:
Hey @niyobobo, your codes are working well but you are destructuring this.props in 2 lines 19 and 21, please do it in once for all those variables.

Reply:
Thank you @fridolinho for the comment, but the logic behind is this is that when a response comes to confirm that your response removed, it goes directly to the initial state that's why I need to destructure it and see what is there.

Reply:
Okay Got it.